source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -3872,6 +3890,83 @@ SwiftLanguageRuntime::GetBridgedSyntheticChildProvider(ValueObject &valobj) {
   return nullptr;
 }
 
+void SwiftLanguageRuntime::WillStartExecutingUserExpression() {
+  std::lock_guard<std::mutex> lock(m_active_user_expr_mutex);
+  Log *log(GetLogIfAnyCategoriesSet(LIBLLDB_LOG_EXPRESSIONS));
+
+  if (m_active_user_expr_count == 0 &&
+      m_dynamic_exclusivity_flag_addr) {
+    // We're executing the first user expression. Toggle the flag.
+    Error error;
+    size_t bytes_read =
+      m_process->DoReadMemory(*m_dynamic_exclusivity_flag_addr,
+                              &m_original_dynamic_exclusivity_flag_state,
+                              sizeof(m_original_dynamic_exclusivity_flag_state),
+                              error);
+
+    if (bytes_read != sizeof(m_original_dynamic_exclusivity_flag_state)) {
+      if (log)
+        log->Printf("SwiftLanguageRuntime: Unable to read "
+                    "disableExclusivityChecking flag state");
+    } else {
+      bool new_flag_state = true;
+      size_t bytes_written =
+        m_process->DoWriteMemory(
+                              *m_dynamic_exclusivity_flag_addr,
+                              &new_flag_state,
+                              sizeof(m_original_dynamic_exclusivity_flag_state),
+                              error);
+      if (bytes_written != sizeof(m_original_dynamic_exclusivity_flag_state)) {
```

Comment:
Same comment as when reading.  You need to get the sizeof(bool) as above.  Then you should make a Scalar with the value of 1 (scalars don't know about bools, but a C++ bool true is an int with value 1) and you can use WriteScalarToMemory to get the byte swapping right.